src/config.rs

```diff
@@ -64,6 +64,19 @@ impl Config {
         self.server_history_dir().join(id_with_timestamp_extension)
     }
 
+    pub(crate) fn document_name_with_default(&self, document_path: &str) -> String {
```

Comment:
Give some sample input and output of this function as part of documentation of this function.